lib/maze/server.rb

```diff
@@ -35,26 +28,34 @@ class << self
       # @return [String] The UUID attached to all command requests for this session
       attr_reader :command_uuid
 
-      # The intended HTTP status code on a successful request
+      # Sets the status code generator for the HTTP verb given.  If no verb is given then the
+      # generator will be shared across all allowable HTTP verbs.
       #
-      # @return [Integer] The HTTP status code, defaults to 200
-      def status_code(verb=nil)
-        if @status_override_verb
-          override_status = @status_override_verb.eql?(verb)
+      # @param verb [String] HTTP verb
+      def set_status_code_generator(generator, verb = nil)
+        @generators ||= {}
+        if verb.nil?
+          ALLOWED_HTTP_VERBS.each do |allowed_verb|
+            @generators[allowed_verb]&.close
+            @generators[allowed_verb] = generator
+          end
         else
-          override_status = true
+          @generators[verb]&.close
```

Comment:
This doesn't work if you set a generator for all codes, then set another generator for a specific code — you'll close a generator that's still being used

This is what I meant by:

> I think you'll struggle to handle that solely in Server as you'd need to track when you drop the last reference to an existing Generator

Reply:
Discussed IRL and method updated to account for generators still in use.